src/Altinn.Broker/Controllers/FileController.cs

```diff
@@ -1,105 +1,250 @@
+            var file = await _fileRepository.GetFileAsync(fileId);
+            if (file is null)
+            {
+                return NotFound();
+            }
+
+            return Ok(FileStatusOverviewExtMapper.MapToExternalModel(file));
         }
 
-        [HttpPut]
-        public async Task<ActionResult<BrokerFileStatusOverviewExt>> OverwriteFileStreamed(Guid fileId, 
-        string fileName, 
-        string sendersFileReference, 
-        string checksum)
+        /// <summary>
+        /// Get more detailed information about the file upload for auditing and troubleshooting purposes
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet]
+        [Route("{fileId}/details")]
+        public async Task<ActionResult<FileStatusDetailsExt>> GetFileDetails(Guid fileId)
         {
-            await Task.Run(() => 1 == 1);
-            BrokerFileStatusOverviewExt brokerFileStatusOverviewExt = new BrokerFileStatusOverviewExt();
-            return Accepted(brokerFileStatusOverviewExt);
+            var caller = GetCallerFromTestToken(HttpContext);
+            if (string.IsNullOrWhiteSpace(caller))
+            {
+                return Unauthorized();
+            }
+            var file = await _fileRepository.GetFileAsync(fileId);
+            if (file is null)
+            {
+                return NotFound();
+            }
+
+            var fileHistory = await _fileRepository.GetFileStatusHistoryAsync(fileId);
+            var actorEvents = await _fileRepository.GetActorEvents(fileId);
+
+            var fileOverview = FileStatusOverviewExtMapper.MapToExternalModel(file);
+            return new FileStatusDetailsExt()
+            {
+                Checksum = fileOverview.Checksum,
+                FileId = fileId,
+                FileName = fileOverview.FileName,
+                Sender = fileOverview.Sender,
+                FileStatus = fileOverview.FileStatus,
+                FileStatusChanged = fileOverview.FileStatusChanged,
+                FileStatusText = fileOverview.FileStatusText,
+                Metadata = fileOverview.Metadata,
+                Recipients = fileOverview.Recipients,
+                SendersFileReference = fileOverview.SendersFileReference,
+                FileStatusHistory = FileStatusOverviewExtMapper.MapToFileStatusHistoryExt(fileHistory),
+                RecipientFileStatusHistory = FileStatusOverviewExtMapper.MapToRecipientEvents(actorEvents.Where(actorEvents => actorEvents.Actor.ActorExternalId != file.Sender).ToList())
+            };
         }
 
-        [HttpPost]
-        [Route("{fileId}/cancel")]
-        public async Task<ActionResult> CancelFile(Guid fileId, string reasonText)
+        /// <summary>
+        /// Get files that can be accessed by the caller according to specified filters. Result set is limited to 100 files.
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet]
+        public async Task<ActionResult<List<Guid>>> GetFiles()
```

Comment:
Consider adding some simple filters that can be expanded in future;
FileStatus, FromDate, ToDate

Reply:
Added filters, but spun out the actual search algorithm in an independent task:
https://github.com/Altinn/altinn-broker/issues/165